### ISSUE TITLE
demo/dwt_signal_decomposition.py : TypeError: object of type 'NoneType' has no len()

### ISSUE THREAD

**Issue by DSLituiev** [PyWavelets/pywt#140]
Running `demo/dwt_signal_decomposition.py` returns:

```
$python3 dwt_signal_decomposition.py
Traceback (most recent call last):
  File "dwt_signal_decomposition.py", line 67, in <module>
    plot_signal_decomp(data1, 'coif5', "DWT: Signal irregularity")
  File "dwt_signal_decomposition.py", line 42, in plot_signal_decomp
    rec_a.append(pywt.waverec(coeff_list, w))
  File "/Users/dlituiev/external/pywt/pywt/_multilevel.py", line 113, in waverec
    if len(a) == len(d) + 1:
TypeError: object of type 'NoneType' has no len()
```

using freshly cloned repo from github

**Comment by grlee77** [comment]
This seems to be a bug in `waverec`.  It is no longer handling cases where some of the coefficients are None.   #143 is a proposed fix for this

**Comment by grlee77** [comment]
This was resolved in #143
